astropy/io/fits/hdu/hdulist.py

```diff
@@ -1068,31 +1061,6 @@ def _try_while_unread_hdus(self, func, index, *args, **kwargs):
         HDUList, but additional arguments may be passed on.
         """
 
-        # TODO: This is a temporary hack for reporting the deprecation
-        # warning on the pre-lazy-loading behavior of allowing HDUs to
-        # be accessed after the file was closed.  It can be removed once
-        # the deprecation period has passed.
-        if self._file and self._file.closed and self._total_read is not None:
-            if index > self._total_read - 1:
-                warnings.warn(textwrap.dedent(
-                    """\
-                    Accessing an HDU after an HDUList is closed, where
-                    that HDU was no read while the HDUList was open
-                    is deprecated.  That is, you did something like:
-
-                        >>> hdulist.close()
-                        >>> print(hdulist[2].header)
-
-                    even though hdulist[2] had not been read yet.  Instead
-                    do:
-
-                        >>> print(hdulist[2].header)
-                        >>> hdulist.close()
-
-                    or open the file with lazy_load_hdus=False to read all
-                    the HDUs into memory immediately upon opening the file.
-                    """), AstropyDeprecationWarning)
-
```

Comment:
My feeling is that this should only be removed **if and only if** #6082 is merged. So I would suggest not removing it in here.